### ISSUE TITLE
has_virtuoso() is FALSE without running vos_start() first

### ISSUE THREAD

**Issue by allenbaron** [ropensci/virtuoso#35]
I used `vos_install()` on MacOS to install virtuoso with no custom modifications, have imported data into the database, and used `vos_query()` for various queries without any errors. However, I noticed after closing & restarting my project in RStudio if I run `has_virtuoso()` the response is `FALSE`. If I run `vos_start()` and then `has_virtuoso()` the response is then `TRUE`. Is this the expected behavior of `has_virtuoso()`?

<details>
<summary> SessionInfo </summary>

```
R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 11.6

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] virtuoso_0.1.6

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7      magrittr_2.0.1  hms_1.1.1       odbc_1.3.2     
 [5] rappdirs_0.3.3  bit_4.0.4       R6_2.5.1        rlang_0.4.12   
 [9] fansi_0.5.0     blob_1.2.2      tools_4.1.1     utf8_1.2.2     
[13] DBI_1.1.1       ellipsis_0.3.2  bit64_4.0.5     digest_0.6.28  
[17] tibble_3.1.5    lifecycle_1.0.1 crayon_1.4.1    processx_3.5.2 
[21] ini_0.3.1       vctrs_0.3.8     fs_1.5.0        ps_1.6.0       
[25] curl_4.3.2      compiler_4.1.1  pillar_1.6.4    renv_0.14.0    
[29] pkgconfig_2.0.3
```

</details>

**Comment by cboettig** [comment]
@allenbaron I'm right there with you that it does not really seem to be the behavior you would expect.  You can see that `vos_start()` first updates the path before calling `has_virtuoso()`, and apparently even has a note about Mac not persisting this path between restarts:


https://github.com/ropensci/virtuoso/blob/7555e052c5e972b9d9fbfcb020adc9c2558d773f/R/vos_start.R#L48-L57


Hmm... This suggests we might want to just call `vos_set_path()`, but unfortunately that creates a problem -- `has_virtuoso()` is actually also called by `vos_set_path()` on order to not mess with paths when virtuoso is already configured:

https://github.com/ropensci/virtuoso/blob/7555e052c5e972b9d9fbfcb020adc9c2558d773f/R/vos_install.R#L59-L66


So yeah, might be better if we had called `has_virutuoso` something like `virtuoso_on_path` (since that is what it actually checks for; that the executable binary is on your path), and then `has_virtuoso()` could call the set_path function first.  I'll try and draft a patch along those lines.    Meanwhile, `virtuoso:::vos_set_path()` should also work.  

Thanks for the bug report!

**Comment by cboettig** [comment]
@allenbaron should be fixed on CRAN now (v0.1.7)

**Comment by allenbaron** [comment]
Thanks for the quick response!
